ui/components/app/simulation-details/useBalanceChanges.ts

```diff
@@ -0,0 +1,175 @@
+import { Hex } from '@metamask/utils';
+import { useSelector } from 'react-redux';
+import { createSelector } from 'reselect';
+import {
+  SimulationBalanceChange,
+  SimulationData,
+  SimulationTokenBalanceChange,
+  SimulationTokenStandard,
+} from '@metamask/transaction-controller';
+import { useAsyncResultStrict } from '../../../hooks/useAsyncResult';
+import { getTokenStandardAndDetails } from '../../../store/actions';
+import { TokenStandard } from '../../../../shared/constants/transaction';
+import { Numeric } from '../../../../shared/modules/Numeric';
+import { getConversionRate } from '../../../ducks/metamask/metamask';
+import {
+  getConfirmationExchangeRates,
+  getTokenExchangeRates,
+} from '../../../selectors';
+import {
+  Amount,
+  BalanceChange,
+  FIAT_UNAVAILABLE,
+  NativeAssetIdentifier,
+  TokenAssetIdentifier,
+} from './types';
+
+const NATIVE_ASSET: NativeAssetIdentifier = { standard: TokenStandard.none };
+const NATIVE_DECIMALS = 18;
+const ERC20_DEFAULT_DECIMALS = 18;
+
+// Converts a SimulationTokenStandard to a TokenStandard
+const convertStandard = (standard: SimulationTokenStandard) => {
+  switch (standard) {
+    case SimulationTokenStandard.erc20:
+      return TokenStandard.ERC20;
+    case SimulationTokenStandard.erc721:
+      return TokenStandard.ERC721;
+    case SimulationTokenStandard.erc1155:
+      return TokenStandard.ERC1155;
+    default:
+      throw new Error(`Unknown token standard: ${standard}`);
+  }
+};
+
+// Calculates the asset amount based on the balance change and decimals
+const getAssetAmount = (
+  { isDecrease: isNegative, difference: quantity }: SimulationBalanceChange,
+  decimals: number,
+): Amount => {
+  const numeric = Numeric.from(quantity, 16)
+    .times(isNegative ? -1 : 1, 10)
```

Comment:
Is this what we added the `abs` method for?

Reply:
In that case, are we doubling up on the `+` and `-` as does the `AmountPill` also add those?

Reply:
The Figma for the AmountPill has a space between the the sign and the number:

<img width="89" alt="image" src="https://github.com/MetaMask/metamask-extension/assets/507015/a1926f7a-a09a-4640-b138-63bc030f11bb">

Also, it shows a "+" sign for positive changes:

<img width="65" alt="image" src="https://github.com/MetaMask/metamask-extension/assets/507015/212b6b6b-589e-446f-a4d3-5ff3e4954cd0">


Because of that, the AmountPill has to render the amount something like this:
```
const formattedAmount = `${sign} ${amount.numeric.abs().toString()}`
```